app/routes/_landing.($lang).docs/route.tsx

```diff
@@ -51,37 +59,79 @@ const organizeData = (docs: documentation[]): LinksGroupProps[] => {
       link: doc.slug || "",
       slug: doc.slug || "",
       links: findChildren(doc.id, docs),
+      hasParent: false,
     }))
 }
 
+function flattenTree(tree: LinksGroupProps[]) {
```

Comment:
I suggest creating a utils file that would contain all of the utility functions we're using for this route. 
Also, I see a few improvements here for the overall implementation of this function keeping modularity and immutability in mind. 

Instead of directly mutating the array we and using loops, we can utilize es6 functions (flatMap). By utilizing flatMap the code becomes more concise and eliminates the need for pushing to and mutating the array. 

Applying the above, the code can look like.. 

```
function flattenTree(tree: LinksGroupProps[]): LinksGroupProps[] {
  return tree.flatMap(dfs);

  function dfs(node: LinksGroupProps): LinksGroupProps[] {
    return [node, ...node.links.flatMap(dfs)];
  }
}

```